src/Controller/EventController.php

```diff
@@ -58,6 +67,48 @@ public function new(Request $request): Response
         ]);
     }
 
+    /**
+     * Upload image to library, add unique name and add the image to the event
+     * @Route("/new/{event}/addposter", name="event_add_poster", methods={"GET","POST"})
+     * @IsGranted("ROLE_ADMIN")
+     */
+    public function newWithPoster(Request $request, FileUploader $fileUploader, Event $event): Response
+    {
+        $poster = new Poster();
+        $form = $this->createForm(PosterType::class, $poster);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            /** @var UploadedFile $posterFile */
+            $posterFile = $form->get('poster_img')->getData();
+            try {
+                $posterSlug = $fileUploader->upload($posterFile, $poster->getFileName());
+            } catch (IniSizeFileException | FormSizeFileException $e) {
+                $this->addFlash('warning', 'Votre fichier est trop lourd, il ne doit pas dépasser 1Mo.');
+                return $this->redirectToRoute('member_add_poster');
+            } catch (ExtensionFileException $e) {
+                $this->addFlash('warning', 'Le format de votre fichier n\'est pas supporté.
+                    Votre fichier doit être au format jpeg, jpg ou png.');
+                return $this->redirectToRoute('member_add_poster');
+            } catch (PartialFileException | NoFileException | CannotWriteFileException $e) {
+                $this->addFlash('warning', 'Fichier non enregistré, veuillez réessayer.
+                    Si le problème persiste, veuillez contacter l\'administrateur du site');
+                return $this->redirectToRoute('member_add_poster');
+            }
+            $poster->setSlug($posterSlug);
+            $entityManager = $this->getDoctrine()->getManager();
```

Comment:
You should inject the EntityManagerInterface as a parameter

Reply:
Yes probably I should have used a parameter ...